internals/feature_helpers.py

```diff
@@ -312,63 +312,149 @@ def get_in_milestone(milestone: int,
+        ndb.OR(Stage.stage_type == STAGE_BLINK_SHIPPING,
+            Stage.stage_type == STAGE_PSA_SHIPPING,
+            Stage.stage_type == STAGE_FAST_SHIPPING,
+            Stage.stage_type == STAGE_DEP_SHIPPING))
+    q = q.filter()
+    desktop_shipping_future = q.fetch_async()
+
+    # Shipping stages with a matching android shipping milestone
+    # but no desktop milestone.
+    q = Stage.query(Stage.milestones.android_first == milestone,
+        Stage.milestones.desktop_first == None,
+        ndb.OR(Stage.stage_type == STAGE_BLINK_SHIPPING,
+            Stage.stage_type == STAGE_PSA_SHIPPING,
+            Stage.stage_type == STAGE_FAST_SHIPPING,
+            Stage.stage_type == STAGE_DEP_SHIPPING))
+    android_only_shipping_future = q.fetch_async()
+
+    # Origin trial stages (Desktop) in this milestone.
+    q = Stage.query(Stage.milestones.desktop_first == milestone,
+        ndb.OR(Stage.stage_type == STAGE_BLINK_ORIGIN_TRIAL,
+            Stage.stage_type == STAGE_FAST_ORIGIN_TRIAL,
+            Stage.stage_type == STAGE_DEP_DEPRECATION_TRIAL))
+    desktop_origin_trial_future = q.fetch_async()
+
+    # Origin trial stages (Android) in this milestone.
+    q = Stage.query(Stage.milestones.android_first == milestone,
+        Stage.milestones.desktop_first == None,
+        ndb.OR(Stage.stage_type == STAGE_BLINK_ORIGIN_TRIAL,
+            Stage.stage_type == STAGE_FAST_ORIGIN_TRIAL,
+            Stage.stage_type == STAGE_DEP_DEPRECATION_TRIAL))
+    android_origin_trial_future = q.fetch_async()
+
+    # Origin trial stages (Webview) in this milestone.
+    q = Stage.query(Stage.milestones.webview_first == milestone,
+        Stage.milestones.desktop_first == None,
+        ndb.OR(Stage.stage_type == STAGE_BLINK_ORIGIN_TRIAL,
+            Stage.stage_type == STAGE_FAST_ORIGIN_TRIAL,
+            Stage.stage_type == STAGE_DEP_DEPRECATION_TRIAL))
+    webview_origin_trial_future = q.fetch_async()
+
+    # Dev trial stages (Desktop) in this milestone.
+    q = Stage.query(Stage.milestones.desktop_first == milestone,
+        ndb.OR(Stage.stage_type == STAGE_BLINK_DEV_TRIAL,
+            Stage.stage_type == STAGE_PSA_DEV_TRIAL,
+            Stage.stage_type == STAGE_FAST_DEV_TRIAL,
+            Stage.stage_type == STAGE_DEP_DEV_TRIAL))
+    desktop_dev_trial_future = q.fetch_async()
+
+    # Dev trial stages (Android) in this milestone.
+    q = Stage.query(Stage.milestones.android_first == milestone,
+        Stage.milestones.desktop_first == None,
+        ndb.OR(Stage.stage_type == STAGE_BLINK_DEV_TRIAL,
+            Stage.stage_type == STAGE_PSA_DEV_TRIAL,
+            Stage.stage_type == STAGE_FAST_DEV_TRIAL,
+            Stage.stage_type == STAGE_DEP_DEV_TRIAL))
+    android_dev_trial_future = q.fetch_async(None)
+
+    # Wait for all futures to complete and collect unique feature IDs.
+    desktop_shipping_ids = list(set(
+        [s.feature_id for s in desktop_shipping_future.result()]))
```

Comment:
You can save a few characters of syntax by using python3 set creation syntax:
list({
   s.feature_id for s in desktop_shipping_future.result()})

Reply:
I tried this, but then realized that this format will try to hash the list rather than unpack it, so it requires a `*` to preface the list. I think it's a bit harder to read what is happening this way, but I went with the change.